lib/dependencies/ContextDependencyTemplateAsRequireCall.js

```diff
@@ -2,38 +2,41 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function ContextDependencyTemplateAsRequireCall() {}
-module.exports = ContextDependencyTemplateAsRequireCall;
+"use strict";
 
-ContextDependencyTemplateAsRequireCall.prototype.apply = function(dep, source, outputOptions, requestShortener) {
-	var comment = "";
-	if(outputOptions.pathinfo) comment = "/*! " + requestShortener.shorten(dep.request) + " */ ";
-	var containsDeps = dep.module && (dep.module.dependencies && dep.module.dependencies.length > 0);
-	var isAsync = dep.module && dep.module.async;
-	if(dep.module && (isAsync || containsDeps)) {
-		if(dep.valueRange) {
-			if(Array.isArray(dep.replaces)) {
-				for(var i = 0; i < dep.replaces.length; i++) {
-					var rep = dep.replaces[i];
-					source.replace(rep.range[0], rep.range[1] - 1, rep.value)
+class ContextDependencyTemplateAsRequireCall {
+
+	apply(dep, source, outputOptions, requestShortener) {
+		let comment = "";
+		if(outputOptions.pathinfo) comment = "/*! " + requestShortener.shorten(dep.request) + " */ ";
+		const containsDeps = dep.module && (dep.module.dependencies && dep.module.dependencies.length > 0);
```

Comment:
I wonder why this had parens; `&&` is indeed left-associative, but due to short-circuiting the order of operations is still the same regardless of the parens